crates/bws/src/state.rs

```diff
@@ -1,17 +1,27 @@
 use std::path::PathBuf;
 
-use color_eyre::eyre::Result;
+use color_eyre::eyre::{bail, Result};
+use directories::BaseDirs;
+
+pub(crate) const DEFAULT_STATE_DIRECTORY: &str = ".config/bws/state";
 
 pub(crate) fn get_state_file(
     state_dir: Option<PathBuf>,
     access_token_id: String,
-) -> Result<Option<PathBuf>> {
-    if let Some(mut state_dir) = state_dir {
-        std::fs::create_dir_all(&state_dir)?;
-        state_dir.push(access_token_id);
+) -> Result<PathBuf> {
+    let mut state_dir = match state_dir {
+        Some(state_dir) => state_dir,
+        None => {
+            if let Some(base_dirs) = BaseDirs::new() {
+                base_dirs.home_dir().join(DEFAULT_STATE_DIRECTORY)
+            } else {
+                bail!("A valid home directory doesn't exist");
```

Comment:
Maybe it would be good option to fallback to opt out automatically when there was a problem with the state directory access in any way ?
We could maybe display a warn instead ?

Reply:
That's a great idea! Fixed here: https://github.com/bitwarden/sdk/pull/930/commits/c520d1fadbd84d2b1f879b0c74c3df280d26b62b